src/commands/createNewProject/IProjectCreator.ts

```diff
@@ -3,21 +3,46 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { OutputChannel } from "vscode";
+import { TelemetryProperties } from "vscode-azureextensionui";
+import { IUserInterface } from "../../IUserInterface";
 import { localize } from "../../localize";
-import { extensionPrefix, ProjectRuntime, TemplateFilter } from "../../ProjectSettings";
+import { extensionPrefix, ProjectRuntime, promptForProjectRuntime, TemplateFilter } from "../../ProjectSettings";
+import { functionRuntimeUtils } from "../../utils/functionRuntimeUtils";
 
-export interface IProjectCreator {
-    deploySubpath?: string;
-    runtime: ProjectRuntime;
-    templateFilter: TemplateFilter;
+export abstract class ProjectCreatorBase {
+    public deploySubpath: string = '';
+    public abstract templateFilter: TemplateFilter;
+
+    protected readonly functionAppPath: string;
+    protected readonly outputChannel: OutputChannel;
+    protected readonly ui: IUserInterface;
+    protected readonly telemetryProperties: TelemetryProperties;
+
+    constructor(functionAppPath: string, outputChannel: OutputChannel, ui: IUserInterface, telemetryProperties: TelemetryProperties) {
+        this.functionAppPath = functionAppPath;
+        this.outputChannel = outputChannel;
+        this.ui = ui;
+        this.telemetryProperties = telemetryProperties;
+    }
+
+    public async getRuntime(): Promise<ProjectRuntime> {
+        // tslint:disable-next-line:strict-boolean-expressions
+        return await functionRuntimeUtils.tryGetLocalRuntimeVersion() || await promptForProjectRuntime(this.ui);
+    }
+
+    public getLaunchJson(): {} | undefined {
+        return undefined;
```

Comment:
Is this always supposed to default to undefined?

Reply:
Yep - only like 3 out of 10-ish languages actually support a launch.json (granted those 3 languages are the 3 we care about most - C#, Java, and JavaScript. The rest are just preview langagues)